### ISSUE TITLE
Expo EAS Android build failure: Could not find io.adapty.internal:crossplatform:2.10.3

### ISSUE THREAD

**Issue by stevennhnk113** [adaptyteam/AdaptySDK-React-Native#115]
### Description ###
- I am adding Adapty package to my Expo Managed application and I am running into issue during running EAS build.

### Details ###
Errors at step `Run gradlew`:
```
* What went wrong:

Could not determine the dependencies of task ':app:compileDebugJavaWithJavac'.

> Could not resolve all task dependencies for configuration ':app:debugCompileClasspath'.

   > Could not find io.adapty.internal:crossplatform:2.10.3.

     Required by:

         project :app > project :react-native-adapty

* Try:

> Run with --stacktrace option to get the stack trace.

> Run with --info or --debug option to get more log output.

> Run with --scan to get full insights.

> Get more help at https://help.gradle.org.

BUILD FAILED in 3m 57s

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.3/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

15 actionable tasks: 15 executed

Error: Gradle build failed with unknown error. See logs for the "Run gradlew" phase for more information.
```

Notable error:
`Could not find io.adapty.internal:crossplatform:2.10.3`

default build tools using by EAS:
```
Android gradle plugin: 8.1.1
Gradle: 8.3
```

dependencies:
```
{
  "dependencies": {
    "expo": "~50.0.14",
    "expo-build-properties": "~0.11.1",
    "expo-dev-client": "~3.3.11",
    "react": "18.2.0",
    "react-native": "0.73.6",
    "react-native-adapty": "^2.10.0",
  },
}

```

### What I have tried ###
- Downgrade Adapty package to `2.9.8`

### Version

v2.10.0

### What platforms are you seeing the problem on?

Android

### System info

```yml
Getting Permission denied, even with `sudo`. Possible due to using EXPO?
```

**Comment by vladd-g** [comment]
@stevennhnk113 thank you for the report! Could you please try [this solution](https://github.com/adaptyteam/AdaptySDK-React-Native/issues/80#issuecomment-1694675411)?

Also it looks similar to [this issue](https://github.com/adaptyteam/AdaptyUI-React-Native/issues/8)

**Comment by stevennhnk113** [comment]
Thanks for the help, but it is still not working and getting the exact same error. 
- I have a monorepo setup, so I gave this a try:
```
"android": {
  "extraMavenRepos": [
    "$rootDir/../../../node_modules/react-native-adapty/lib/android/localMaven"
  ]
}
```

- I also tried this as well:
```
"android": {
  "extraMavenRepos": [
    "./node_modules/react-native-adapty/lib/android/localMaven"
  ]
}
```

**Comment by vladd-g** [comment]
@stevennhnk113 nothing else from the links above worked either, did it?

**Comment by KristianLentino99** [comment]
I have the same problem

**Comment by stevennhnk113** [comment]
@vladd-g Unfortunately, none of the solution work. My project use Expo Managed, so some of the solutions are not applicable such as this [example](https://github.com/adaptyteam/AdaptySDK-React-Native/issues/80#issuecomment-1636871965)

 Lets me know if you need more information.

**Comment by MartinGambit** [comment]
Any solution? I am having the same issue and none of the suggestions above helped me.
I am using RN expo bare workflow

**Comment by benimbolat** [comment]
I had the same problem. Anyone found a solution?

**Comment by mpanac** [comment]
I couldn't make it work and the team refused to help so i had to switch to a revenueCat.

> I had the same problem. Anyone found a solution?

**Comment by stevennhnk113** [comment]
> I had the same problem. Anyone found a solution?

I switched to revenueCat as well.

**Comment by vladd-g** [comment]
@benimbolat @mpanac [this one](https://github.com/adaptyteam/AdaptySDK-React-Native/issues/80#issuecomment-1685038260) in your case didn't help either? or [this](https://github.com/adaptyteam/AdaptySDK-React-Native/issues/119#issuecomment-2112281520), like [here](https://github.com/adaptyteam/AdaptySDK-React-Native/issues/80#issuecomment-1636871965)?

additionally, maybe it requires [prebuilding](https://github.com/adaptyteam/AdaptyUI-React-Native/issues/8#issuecomment-1971071876) at some stage

**Comment by benimbolat** [comment]
Hi @vladd-g,

Thank you for your message. When I encountered this problem, I solved the problem by buying eas build. Today I tried again and my problem was solved with prebuild.

Thank you for everything

```
    "@adapty/react-native-ui": "^2.11.1",
    "react-native-adapty": "^2.11.3",
    "expo": "~51.0.36",    
    "react-native": "0.74.5",
```

**_Folder =>  android > build.gradle_** 
```
 allprojects {
    repositories { 

         //... Other repositories
        maven { url '../../node_modules/@adapty/react-native-ui/android/localMaven' }
        maven { url '../../node_modules/react-native-adapty/lib/android/localMaven' }

      }
  }
```

**Comment by vladd-g** [comment]
The problem should be completely resolved in v3.x

**Comment by necmettindev** [comment]
> The problem should be completely resolved in v3.x

Still getting same error

**Comment by vladd-g** [comment]
@necmettindev did you try [prebuilding](https://docs.expo.dev/workflow/prebuild/)?

**Comment by necmettindev** [comment]
> @necmettindev did you try [prebuilding](https://docs.expo.dev/workflow/prebuild/)?

I did it , this is not related to the prebuild. Your internal maven is not recognised by expo. (Btw everything works in iOS build)

<img width="447" alt="image" src="https://github.com/user-attachments/assets/af0f03b5-3a7a-4ed4-b519-9c30198a99d8">

**Comment by vladd-g** [comment]
but there is no internal maven anymore since 3.x

did you sync Android gradle project?

**Comment by necmettindev** [comment]
> but there is no internal maven anymore since 3.x
> 
> did you sync Android gradle project?

yes, if you want we can do pair for quick solution, I came from revenuecat and don't want to return revenuecat. (I like adapty features)

**Comment by vladd-g** [comment]
sure, let's do it!

is it a release build on your screenshot? if so, do you have any problems with a debug build?

**Comment by necmettindev** [comment]
Can we continue on discord or telegram? @vladd-g

**Comment by vladd-g** [comment]
@necmettindev sure, how can I find you there?

**Comment by necmettindev** [comment]
@vladd-g  could you send your discord channel?

**Comment by vladd-g** [comment]
@necmettindev you can share any convenient contact at support@adapty.io

**Comment by necmettindev** [comment]
> @necmettindev you can share any convenient contact at [support@adapty.io](mailto:support@adapty.io)

sent

**Comment by mustafasenturk** [comment]
any news?

**Comment by vladd-g** [comment]
@mustafasenturk do you have any errors with v3.x?

**Comment by troberts-28** [comment]
I'm running into similar (though not the same) issues with v3.x on Expo SDK 52 @vladd-g.

```
"expo": "52.0.7",
"react": "18.3.1",
"react-native": "0.76.2",
"react-native-adapty": "3.0.1",
"@adapty/react-native-ui": "3.0.1",
```

The project builds fine but fails on loading the app with:

```
Failed resolution of: Lcom/adapty/internal/corssplatform/CrossplatformHelper;

Stack trace:
init$crossplatform_ui_release
Dependencies.kt

init
CrossplatformUiHelper.kt

...
```

Also had an issue with v2.11.0 on Expo SDK 52. Any help would be much appreciated - this is currently blocking our upgrade to Expo SDK 52.

**Comment by vladd-g** [comment]
hi @troberts-28

thanks for the report, working on it! could you try disabling the expo auto-linking for now?

**Comment by troberts-28** [comment]
Appreciate the quick response and glad to know you're working on a fix!

Will try disabling auto-linking for now. Should I disable it for both the @adapty/react-native-ui and react-native-adapty?

**Comment by vladd-g** [comment]
Could you try it only for react-native-adapty?

**Comment by troberts-28** [comment]
Tried that and got the same issue. Not 100% sure I disabled it correctly though... I followed the instructions here: https://docs.expo.dev/modules/autolinking/#exclude

```
// react-native.config.js
module.exports = {
    dependencies: {
        "react-native-adapty": {
            platforms: {
                android: null,
            },
        },
    },
};
```

**Comment by vladd-g** [comment]
@troberts-28 could you please try `3.0.2-canary.0`, without disabling auto-linking?

**Comment by troberts-28** [comment]
Hey @vladd-g, we've paused the Expo SDK 52 upgrade for now so not going to be able to try this immediately. Will give you a shout when we do.

**Comment by mbilalbark** [comment]
Hi @vladd-g, I tried your solution and installed version `3.0.2-canary.0`. It works on the emulator, but I am having issues with the production build.

**Comment by johannbuscail** [comment]
@vladd-g Any update ?

**Comment by vladd-g** [comment]
@johannbuscail could you please share details on the errors you have?
